addon/models/osf-model.js

```diff
@@ -33,24 +33,35 @@ export default DS.Model.extend({
     ready() {
         this._super(...arguments);
         this.set('_dirtyRelationships', Ember.Object.create({}));
-
-        this.eachRelationship((relationship, meta) => {
-            let rel = relationship;
-            this.get(rel).then(() => {
-                var watch = rel;
-                if (meta.kind === 'hasMany') {
-                    watch = `${rel}.[]`;
+    },
+    save() {
+        this.eachRelationship((rel, meta) => {
+            if (meta.kind === 'hasMany') {
+                let relation = this.hasMany(rel).hasManyRelationship;
+                if (relation.record.isLoaded()) {
+                    if (
+                        (relation.canonicalState.filter(record => record && (Object.keys(record.changedAttributes()).length > 0 || record.isNew())).length > 0) ||
+                        relation.canonicalMembers.size !== relation.members.size
+                    ) {
+                        let key = `_dirtyRelationships.${rel}`;
+                        this.set(key, true);
+                    }
                 }
-                this.addObserver(rel, () => {
-                    var key = `_dirtyRelationships.${rel}`;
-                    this.set(key, !Ember.isEmpty(this.get(key)));
-                    // Checks is_embedded flag set in osf-serializer.
-                    if (this.get('relationshipLinks') && this.get('relationshipLinks')[rel] && this.get('relationshipLinks')[rel].is_embedded) {
-                        this.set(key, false);
-                        this.get('relationshipLinks')[rel].is_embedded = false;
+            } else if (meta.kind === 'belongsTo') {
+                let relation = this.belongsTo(rel).belongsToRelationship;
+                if (relation.record.isLoaded()) {
+                    var record = relation.members.list[0];
```

Comment:
The one edge case is if someone adds an existing object, and removes another previous to a save, without changing any of their attributes, which we can avoid by iterating through both canonical and members and doing a set comparison of the ids, but I don't know if that is necessary